### PR TITLE
duplicate code moved to single place

### PR DESCRIPTION
**Pull request by guptasourabh**
changes -:
Duplicate code moved to single place.
Removed redundant array null check in for loop and initialization of array size.

### CODE REVIEW

File: okhttp/src/main/java/okhttp3/CacheControl.java [hunk 1]
```diff
@@ -350,5 +342,11 @@ public Builder noTransform() {
     public CacheControl build() {
       return new CacheControl(this);
     }
+
+    private int getMaxValueInSeconds(int valueToConvert, TimeUnit timeUnit) {
```
Comment (NightlyNexus) [inline okhttp/src/main/java/okhttp3/CacheControl.java]: static.

File: okhttp/src/main/java/okhttp3/CacheControl.java [hunk 2]
```diff
@@ -350,5 +342,11 @@ public Builder noTransform() {
...
+      long valueInSeconds = timeUnit.toSeconds(valueToConvert);
+      return valueInSeconds > Integer.MAX_VALUE
+              ? Integer.MAX_VALUE : (int) valueInSeconds;
```
Comment (NightlyNexus) [inline okhttp/src/main/java/okhttp3/CacheControl.java]: `return Math.max(Integer.MAX_VALUE, (int) valueInSeconds);` if touching this code anyway?

Comment (JakeWharton) [inline okhttp/src/main/java/okhttp3/CacheControl.java]: That behavior is not the same.
```
Math.min(Integer.MAX_VALUE, (int) (1L + Integer.MAX_VALUE))
===> -2147483648
```

Comment (NightlyNexus) [inline okhttp/src/main/java/okhttp3/CacheControl.java]: oh, haha, right, yeah. 🐑

**Comment by swankjesse** [comment]
Could you complete Square’s contributor agreement?
https://github.com/square/okhttp/blob/master/.github/CONTRIBUTING.md

**Comment by guptasourabh** [comment]
@NightlyNexus changes done
@swankjesse completed square contributor agreement.

**Comment by swankjesse** [comment]
I know that small contributions like these are well-intentioned, but we’re more interested in more dramatic improvements: bug fixes, measurable optimizations, and documentation improvements.

As is it seems like the motivation of this contribution is to make a contribution, and that distracts us from other improvements.

**Pull request closed without merging**